app/services/theme.js

```diff
@@ -31,21 +36,54 @@ export default Ember.Service.extend({
             .findRecord('preprint-provider', id);
     }),
 
+    // If we're using a branded provider
     isProvider: Ember.computed('id', function() {
-        const id = this.get('id');
-        return id && id !== 'osf';
+        return this.get('id') !== 'osf';
+    }),
+
+    // If we're using a branded provider and not under a branded domain (e.g. /preprints/<provider>)
+    isSubRoute: Ember.computed('isProvider', 'isDomain', function() {
+        return this.get('isProvider') && !this.get('isDomain');
+    }),
+
+    pathPrefix: Ember.computed('isProvider', 'isDomain', 'id', function() {
+        let pathPrefix = '/';
+
+        if (!this.get('isDomain')) {
+            pathPrefix += 'preprints/';
+
+            if (this.get('isProvider')) {
+                pathPrefix += `${this.get('id')}/`;
```

Comment:
wait, just realized that there's also something called `isProvider`. When would `isDomain` and `isProvider` not both be the same?